wrapper/src/main/java/software/amazon/jdbc/plugin/staledns/AuroraStaleDnsHelper.java

```diff
@@ -170,7 +179,7 @@ public void notifyNodeListChanged(final Map<String, EnumSet<NodeChangeOptions>>
   }
 
   private HostSpec getWriter() {
-    for (final HostSpec host : this.pluginService.getHosts()) {
+    for (final HostSpec host : this.pluginService.getAllHosts()) {
```

Comment:
Earlier when we were walking through the code trying to decide where to call getHosts vs getAllHosts we decided to use getAllHosts here because we are trying to identify the writer. Additionally, the beginning of this method checks whether we are using a writer cluster URL and returns early (before calling getWriter) if we are not. So we will not get to this line if we are connecting to a custom endpoint URL.